cmd/soroban-rpc/internal/config/config_test.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"fmt"
+	"runtime"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"github.com/stellar/go/network"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func ExampleConfig() {
+	var cfg Config
+	var err error
+
+	// If you want to load from cli flags, you must call Bind
+	cfg.Bind()
+
+	// Load values from: defaults, env vars, cli flags, then config file
+	// Priority: defaults < config file < env vars < cli flags
+	err = cfg.SetValues()
+	if err != nil {
+		panic(err)
+	}
+
+	// Ensure that what we parsed makes sense
+	err = cfg.Validate()
+	if err != nil {
+		// This is commented in this example, because the default values are
+		// missing some required fields, so the default config we parsed above is
+		// invalid.
+		// panic(err)
+	}
+
+	// Use the values
+	fmt.Println(cfg.Endpoint)
+
+	// Output: localhost:8000
+}
+
+func TestConfigLoadDefaults(t *testing.T) {
+	// Set up a default config
+	cfg := Config{}
+	require.NoError(t, cfg.loadDefaults())
+
+	// Check that the defaults are set
+	assert.Equal(t, network.FutureNetworkPassphrase, cfg.NetworkPassphrase)
+	assert.Equal(t, uint(runtime.NumCPU()), cfg.PreflightWorkerCount)
+	// TODO: Check other defaults
```

Comment:
either do check other defaults or remove the "TODO".
( unless you have plan to follow up on this one )